executor.go

```diff
@@ -124,6 +124,28 @@ func (s *executor) run(mainCtx context.Context,
 				}
 			}()
 
+			// Start a goroutine which will consume state updates
+			// from the server and pipe them into our updates
+			// channel.
+			s.wg.Add(1)
+			go func() {
+				defer s.wg.Done()
+
+				errChan, err := newSwap.subscribeServerUpdates(
```

Comment:
I would move all of this into the swap execution itself. Keep it together. In the future it is likely that we need the server updates for more than just informing the user. For example fast preimage reveal on loop in.

Reply:
I'm unsure that the swap needs to know about this subscription. Since we never want to rely on these messages for swap execution (that would make our setup trusted), they're only really ever going to be for monitoring? Eg payment shards arriving, there's nothing the swap execution can do with that extra info, it's just nice for the user to know. 

I'd say that preimage reveal should be a separate endpoint. Avoid the temptation to overload this one subscription.